sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/spanner/SpannerIO.java

```diff
@@ -163,6 +181,8 @@
 public class SpannerIO {
 
   private static final long DEFAULT_BATCH_SIZE_BYTES = 1024 * 1024; // 1 MB
+  private static final long MAX_NUM_KEYS = (long) 1e6;
+  private static final int DEFAULT_NUM_SAMPLES = 1000;
```

Comment:
<!--thread_id:cc_140938664_t; commit:fad34e7ff681545f907716633ed16e95802b7969; resolved:0-->
<!--section:context-quote-->
> **mairbek** wrote:
> Taken randomly, ideally we would have calculate this based on the size of the input.

<!--section:body-->
Should we add a TODO / file a JIRA ?

Reply:
Done